README.md

```diff
@@ -1,4 +1,114 @@
+#### Rifle - Vampire Limitation
+It **cannot** do many-to-many
+
+#### Use Cases for `Rifle - Vampire`
+* Reliable messaging without responses
+* High performance (500k msgs per second or higher) with zero_copy
+* Process to process communication
+
+#### API
+[[Vampire.h]](https://github.com/LogRhythm/QueueNado/blob/master/src/Vampire.h)
+[[Rifle.h]](https://github.com/LogRhythm/QueueNado/blob/master/src/Rifle.h)
+
+#### Test usage
+[[RifleVampireTests.cpp]](https://github.com/LogRhythm/QueueNado/blob/master/test/RifleVampireTests.cpp)
+
+# Shotgun - Alien
+`Shotgun - Alien` implements the [pub / sub](http://zguide.zeromq.org/page:all#Getting-the-Message-Out) messaging pattern in zmq.
+
+#### Use Cases for `Shotgun - Alien`
+* One to many: one sender communicating with many listeners.
+* Not high performance around 10k msgs a sec. This can be improved by batching many messages together.
+* Process to process communication
+
+#### Known limitations and issues
+* [Slow joiner](http://zguide.zeromq.org/php:chapter5#Representing-State-as-Key-Value-Pairs) issues don't matter or can be worked around
+* All listeners need to see every message for a topic
+
+#### API
+[[Shotgun.h]](https://github.com/LogRhythm/QueueNado/blob/master/src/Shotgun.h)
+[[Alien.h]](https://github.com/LogRhythm/QueueNado/blob/master/src/Alien.h)
+
+#### Test usage
+[[ShotgunAlienTests.cpp]](https://github.com/LogRhythm/QueueNado/blob/master/test/ShotgunAlienTests.cpp)
+
+
+# Headcrab - Crowbar
+`Headcrab - Crowbar` implements [request / reply](http://zguide.zeromq.org/page:all#Ask-and-Ye-Shall-Receive) messaging pattern in zmq.
+
+#### Use Cases for `Headcrab - Crowbar` 
+* Need reply to guarantee message was delivered
+* Know that their will always be a response (otherwise the socket will be broken)
+
+#### Known limitations and issues
+* Lower performance (60k msgs a sec)
+
+[[Headcrab.h]](https://github.com/LogRhythm/QueueNado/blob/master/src/Headcrab.h)
+[[Crowbar.h]](https://github.com/LogRhythm/QueueNado/blob/master/src/Crowbar.h)
+
+#### Test usage
+[[CrowbarHeadcrabTests.cpp]](https://github.com/LogRhythm/QueueNado/blob/master/test/CrowbarHeadcrabTests.cpp)
+
+
+# Harpoon - Kraken
+`Harpoon - Kraken` implements a streaming version of [pub / sub](http://zguide.zeromq.org/page:all#Getting-the-Message-Out). It enables  data streaming from a server to a client. 
+
+
+#### Kraken Purpose:
+The server sending streaming data.
+#### Usage example calls from the API: [[kraken.h]](https://github.com/LogRhythm/QueueNado/blob/master/src/Kraken.h)
+
+* `SendTidalWave()` : Send a data chunk to the client ([[harpoon]](https://github.com/LogRhythm/QueueNado/blob/master/src/Harpoon.h)). The call blocks until there is space available in the queue. Returns `TIMEOUT`, `INTERRUPT`, `CONTINUE` status to indicate the status of the underlying queue.
```

Comment:
I changed the location of `server` to be `publisher` and the Kraken text that had `client` is now referring to it as a `subscriber`